lib/fastlane/plugin/wpmreleasetoolkit/actions/common/buildkite_add_trigger_step_action.rb

```diff
@@ -0,0 +1,165 @@
+# frozen_string_literal: true
+
+require 'open3'
+
+module Fastlane
+  module Actions
+    class BuildkiteAddTriggerStepAction < Action
+      def self.run(params)
+        # Extract parameters
+        pipeline_file = params[:pipeline_file]
+        build_name = File.basename(pipeline_file, '.yml')
+        message = params[:message] || build_name
+        branch = params[:branch] || `git rev-parse --abbrev-ref HEAD`.strip
+        environment = params[:environment] || {}
+        buildkite_pipeline_slug = params[:buildkite_pipeline_slug]
+        async = params[:async]
+        label = params[:label] || ":buildkite: Trigger #{build_name} on #{branch}"
+        depends_on = params[:depends_on]&.then { |v| v.empty? ? nil : Array(v) }
+
+        # Add the PIPELINE environment variable to the environment hash
+        environment = environment.merge('PIPELINE' => pipeline_file)
+
+        # Create the trigger step YAML
+        trigger_yaml = {
+          'steps' => [
+            {
+              'trigger' => buildkite_pipeline_slug,
+              'label' => label,
+              'async' => async,
+              'build' => {
+                'branch' => branch,
+                'message' => message,
+                'env' => environment
+              },
+              'depends_on' => depends_on
+            }.compact,
+          ]
+        }.to_yaml
+
+        # Use buildkite-agent to upload the pipeline
+        _stdout, stderr, status = Open3.capture3('buildkite-agent', 'pipeline', 'upload', stdin_data: trigger_yaml)
+
+        # Check for errors
+        UI.user_error!("Failed to upload pipeline: #{stderr}") unless status.success?
+
+        # Log success
+        UI.success("Added a trigger step to the current Buildkite build to start a new build for #{pipeline_file} on branch #{branch}")
+      end
+
+      def self.description
+        'Adds a trigger step to the current Buildkite pipeline to start a new build from this one'
+      end
+
+      def self.details
+        <<~DETAILS
+          This action adds a `trigger` step to the current Buildkite build, to start a separate build from the current one.
+
+          This is slightly different to `buildkite-agent pipeline upload`-ing the YAML steps of the build directly to the current build,
```

Comment:
Either way sounds about right to me, but I *think* this is the correct form:

```suggestion
          This is slightly different from `buildkite-agent pipeline upload`-ing the YAML steps of the build directly to the current build,
```